datanode/sqlstore/ledger.go

```diff
@@ -191,31 +167,73 @@ func (ls *Ledger) Export(
 		},
 	}
 
-	aggregatedEntries, pageInfo, err := ls.Query(ctx, filter, dateRange, pagination)
+	dynamicQuery, whereQuery, args, err := prepareQuery(filter, dateRange)
+	if err != nil {
+		return err
+	}
+
+	query := fmt.Sprintf("copy (%s %s) to STDOUT (FORMAT csv, HEADER)", dynamicQuery, whereQuery)
+
+	for i, arg := range args {
+		if s, ok := toHexString(arg); ok {
+			arg = s
+		}
+		query = strings.Replace(query, fmt.Sprintf("$%d", i+1), fmt.Sprintf("'%v'", arg), 1)
+	}
```

Comment:
@pscott31 This is a shameful hack... Since `CopyTo` doesn't take any args, I didn't know what else to do so I just replaced the `$1,$2,$3...` bits in the query with actual values and hoped nobody would notice 🫣 🥺

Reply:
Not sure what alternatives we have at this point, as `CopyTo` doesn’t take args. @pscott31 any thoughts?

Reply:
I think we have a potential [bobby tables](https://xkcd.com/327/) situation with this bit of code

strings.Replace(query, fmt.Sprintf("$%d", i+1), fmt.Sprintf("'%v'", arg), 1)

 here right? I think what you want is probably [this sanitising](https://pkg.go.dev/gopkg.in/jackc/pgx.v1#Connection.SanitizeSql) function from pgx

Reply:
I agree with @pscott31 here. I'd also add that `WHERE` clauses re-use parameters, something like:

```
SELECT foo.bar, bar.foo FROM foo JOIN bar ON bar.id = foo.bar_id WHERE foo.something = $1 AND bar.something = $2 OR foo.something_else IN ($1, $2, $3);
```

In which case the `strings.Replace(query, "$%d", replacement, 1)` won't quite cut it.

Reply:
Hey @pscott31 because the method you suggested is from another library version (pgx.v1) I decided instead to copy only the method `SanitizeSql` and modify it a bit to make it cleaner, and go with that. I think it looks better now

Reply:
Also, fwiw, I think raw strings cannot really reach the diver itself, it all goes through those `entities.PartyID` and such types.

Reply:
hi ho, 

the syntax might have changed a bit but I believe it's still exposed in pgxv4 - e.g. here's a bit of our code that does something similar:

```
func dropView(ctx context.Context, conn *pgx.Conn, view pgx.Identifier) error {
	statement := fmt.Sprintf("DROP VIEW IF EXISTS %s CASCADE", view.Sanitize())
	_, err := conn.Exec(ctx, statement)
	if err != nil {
		return fmt.Errorf("dropping view %s: %w", view.Sanitize(), err)
	}
	return nil
}
```

I dunno if you can use this directly or not being in an 'internal' package: 

https://pkg.go.dev/github.com/jackc/pgx/v4/internal/sanitize

I appreciate your point about it probably being safe due to going through partyID types or what have you, but it would be so super duper easy to at some point have some general text field here like 'reference' or something and it would be very embarrassing to fall prey to SQL injection attack :) plus without going and thinking about it and reading the code I wouldn't be sure that we've checked it's a hex string by the time it gets to this query :)

Reply:
Yeah, forgot to mention, it IS in the `internal` package, unfortunately and frustratingly(!)